aten/src/ATen/native/cuda/ScanKernels.cu

```diff
@@ -371,13 +380,16 @@ __host__ void scan_outer_dim(const Tensor& self, Tensor& result,
 
 template <typename scalar_t, class BinaryFunction>
 void scan_innermost_dim(const Tensor& self, Tensor& result, scalar_t init, BinaryFunction binary_op) {
-  int ndim = self.dim();
+  int64_t ndim = self.dim();
   // Treat all outer dimensions as a single dimension.
-  int row_size = self.size(ndim - 1);
-  int num_rows = self.numel() / row_size;
+  int64_t row_size = self.size(ndim - 1);
+  int64_t num_rows = self.numel() / row_size;
 
   dim3 threads(16, 32);
-  dim3 grid(std::min(at::cuda::getCurrentDeviceProperties()->maxGridSize[0], ceil_div(num_rows, int(threads.y))));
+  int64_t maxGridDim = at::cuda::getCurrentDeviceProperties()->maxGridSize[0];
+  dim3 grid(std::min(maxGridDim, ceil_div(num_rows, int64_t{threads.y})));
+
+  check_fits_in_unsigned(num_rows, "Number of rows (self.numel()/self.size(self.dim()-1))");
```

Comment:
and check row_size here?

Reply:
Oops. Had missed that. Thanks for catching that.